tracer/src/Datadog.Trace/Iast/Aspects/System/StringAspects.cs

```diff
@@ -807,6 +905,66 @@ public static string JoinString(string separator, IEnumerable<string> values)
         return result;
     }
 
+#if NET6_0_OR_GREATER
+
+    /// <summary>
+    /// String.Join aspect
+    /// </summary>
+    /// <param name="separator"> sparator </param>
+    /// <param name="values"> values to join </param>
+    /// <returns> Join result </returns>
+    [AspectMethodReplace("System.String::Join(System.String,System.ReadOnlySpan`1<System.String>)")]
+    public static string Join(string separator, ReadOnlySpan<string> values)
+#pragma warning disable DD0005 // Aspect is in incorrect format
+    {
+        return Join(separator, values.ToArray());
```

Comment:
Perhaps this can be done as a follow-up, but can you write a NET6_0_OR_GREATER version of `Join` / `OnStringJoin<T>` to accept and iterate over a `ReadOnlySpan<string>`? I can't see an obvious reason why you couldn't, but maybe I'm missing something

Reply:
I was trying to establish the same thing for `Concat(ReadOnlySpan<string>)` above but I think actually the "trivial" implementation is basically [this](https://github.com/dotnet/runtime/blob/1d1bf92fcf43aa6981804dc53c5174445069c9e4/src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs#L913-L1001) which is far from trivial 😅

Reply:
I agree. But right now these methods will be called very seldom (if anytime at all), and it would imply duplicating code, so I prefer the lesser evil of an extra alloc for the sake of simplicity. But in the future, when these overloads become more "popular" it's something to consider.